File: core/core-enum/src/main/java/com/imstargg/core/enums/BattleMode.java
package com.imstargg.core.enums;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.function.Function;
import java.util.stream.Collectors;

public enum BattleMode {

    NOT_FOUND("notFound"),

    KNOCKOUT("knockout"),
    GEM_GRAB("gemGrab"),
    HEIST("heist"),
    HOT_ZONE("hotZone"),
    SIEGE("siege"),
    BOUNTY("bounty"),
    BRAWL_BALL("brawlBall"),
    DUELS("duels"),
    SOLO_SHOWDOWN("soloShowdown"),
    DUO_SHOWDOWN("duoShowdown"),
    WIPEOUT("wipeout"),
    VOLLEY_BRAWL("volleyBrawl"),
    TROPHY_THIEVES("trophyThieves"),
    BOSS_FIGHT("bossFight"),
    BIG_GAME("bigGame"),
    BASKET_BRAWL("basketBrawl"),
    ROBO_RUMBLE("roboRumble"),
    PAYLOAD("payload"),
    TAKEDOWN("takedown"),
    HUNTERS("hunters"),
    BOT_DROP("botDrop"),
    HOLD_THE_TROPHY("holdTheTrophy"),
    ;

    static {
        Set<String> codes = new HashSet<>();
        for (BattleMode mode : values()) {
            if (!codes.add(mode.getCode())) {
                throw new IllegalStateException("Duplicate code: " + mode.getCode());
            }
        }
    }

    private final String code;

    private static final Map<String, BattleMode> ENUM_BY_CODE = Arrays.stream(BattleMode.values())
            .filter(e -> e != NOT_FOUND)
            .collect(Collectors.toMap(BattleMode::getCode, Function.identity()));

    public static BattleMode find(String code) {
        return ENUM_BY_CODE.getOrDefault(code, NOT_FOUND);
    }

    BattleMode(String code) {
        this.code = code;
    }

    public String getCode() {
        return code;
    }
}

File: admin/imstargg-admin-frontend/src/model/enums/BattleMode.ts
export const BattleMode = {
  NOT_FOUND: 'NOT_FOUND',
  KNOCKOUT: 'KNOCKOUT',
  GEM_GRAB: 'GEM_GRAB',
  HEIST: 'HEIST',
  HOT_ZONE: 'HOT_ZONE',
  SIEGE: 'SIEGE',
  BOUNTY: 'BOUNTY',
  BRAWL_BALL: 'BRAWL_BALL',
  DUELS: 'DUELS',
  SOLO_SHOWDOWN: 'SOLO_SHOWDOWN',
  DUO_SHOWDOWN: 'DUO_SHOWDOWN',
  WIPEOUT: 'WIPEOUT',
  VOLLEY_BRAWL: 'VOLLEY_BRAWL',
  TROPHY_THIEVES: 'TROPHY_THIEVES',
  BOSS_FIGHT: 'BOSS_FIGHT',
  BIG_GAME: 'BIG_GAME',
  BASKET_BRAWL: 'BASKET_BRAWL',
  ROBO_RUMBLE: 'ROBO_RUMBLE',
  PAYLOAD: 'PAYLOAD',
  TAKEDOWN: 'TAKEDOWN',
  HUNTERS: 'HUNTERS',
} as const;

export type BattleModeType = typeof BattleMode[keyof typeof BattleMode];

export const BattleModeValues = Object.keys(BattleMode) as BattleModeType[];